cpp/test/test_geometry.cc

```diff
@@ -212,3 +214,41 @@ BOOST_AUTO_TEST_CASE(check_no_overlap)
   bnd->emplace("bbb", std::set<dealii::types::boundary_id>{1});
   BOOST_CHECK_THROW(cap::Geometry<2>(tria, mat, bnd), std::runtime_error);
 }
+
+// Check that repetitions create a connex triangulation
+BOOST_AUTO_TEST_CASE(check_repetitions)
+{
+  boost::property_tree::ptree geometry_database;
+  boost::property_tree::info_parser::read_info("generate_mesh.info",
+                                               geometry_database);
+  geometry_database.put("n_repetitions", 5);
+  cap::Geometry<2> geo(
+      std::make_shared<boost::property_tree::ptree>(geometry_database),
+      boost::mpi::communicator());
+
+  std::shared_ptr<dealii::distributed::Triangulation<2>> tria =
+      geo.get_triangulation();
+  dealii::DynamicSparsityPattern sparsity_pattern;
+  dealii::GridTools::get_face_connectivity_of_cells(*tria, sparsity_pattern);
+
+  std::unordered_set<unsigned int> cells_done;
+  std::unordered_set<unsigned int> cells_to_do;
+  cells_to_do.insert(0);
+  unsigned int const n_cells = tria->n_active_cells();
+  while (cells_to_do.size() != 0)
+  {
+    unsigned int const current_cell = *cells_to_do.begin();
+    for (unsigned int i = 0; i < n_cells; ++i)
+      if (sparsity_pattern.exists(current_cell, i))
+        if (cells_done.count(i) == 0)
+          cells_to_do.insert(i);
+
+    // The current cell may not be the first element anymore
+    std::unordered_set<unsigned int>::iterator cell_it =
+        cells_to_do.find(current_cell);
+    cells_to_do.erase(cell_it);
+    cells_done.insert(current_cell);
+  }
+
+  BOOST_CHECK(cells_done.size() == n_cells);
+}
```

Comment:
I spent 15min at least to find out what was going on here. You are asking deal.II for a sparsity pattern that connects cells when they have a face in common and you see if you can traverse the entire triangulation. You keep track of the cells you visit in `cells_done` and you check that you did not miss any; because if you did it means that the triangulation as some disjoint (not sure it's the proper term) set of cells. This would catch it if merging two layers fails because of some buggy offset but note that it does not guarantee that they aren't any "holes" in your connectivity. Please add some more comments.